privatelink/aws/dns-endpoints.sh

```
@@ -39,17 +43,23 @@ for nameId in $(aws ec2 describe-availability-zones \
     zonemap[$name]=$id
 done
 
-fmt="  %-25s CNAME %s\n"
 for name in $(aws ec2 describe-vpc-endpoints \
     --vpc-endpoint-ids "$endpoint" \
     --query 'VpcEndpoints[*].DnsEntries[*].[DnsName]' \
     --output text); do
     zoneName=$(echo "$name" | sed -E -e 's/\..*/./' -e 's/^[^-]*-[^-]*-[^-]*-?([^.]*)?\./\1/')
     if [[ -z $zoneName ]]; then
-        # shellcheck disable=SC2059
-        printf "$fmt" "*" "$name"
+        id="*"
     else
-        # shellcheck disable=SC2059
-        printf "$fmt" "*.${zonemap[$zoneName]}" "$name"
+        id="*.${zonemap[$zoneName]}"
     fi
+    dnsrecord[$id]=$name
 done
+
+fmt="  %-25s CNAME %s\n"
+# shellcheck disable=SC2059
+if [[ $zoneid != "*" && -v dnsrecord["*.$zoneid"] ]]; then
```

We should return an error if the user specifies an invalid zoneid.